validation.go

```diff
@@ -149,3 +190,71 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+	multiErr := assignParamDefaults(params, config)
+
+	for pKey, param := range params {
+		if config[pKey] != "" {
+			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, config[pKey]))
+		}
+		multiErr = multierr.Append(multiErr, validateParamValue(pKey, config[pKey], param))
+	}
+	return multiErr
+}
+
+// validateParamValue validates that a configuration value matches all the validations required for the parameter.
+func validateParamValue(key string, value string, param Parameter) error {
+	var multiErr error
+	var err error
+
+	for _, v := range param.Validations {
+		err = v.validate(value)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("error validating %q: %w", key, err))
+		}
+	}
+	return multiErr
+}
+
+// assignParamDefaults fills any empty configuration with its assigned default value
+// returns an error if a parameter is not recognized.
+func assignParamDefaults(params map[string]Parameter, config map[string]string) error {
+	var multiErr error
+	for key, val := range config {
+		if _, ok := params[key]; !ok {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("unrecognized parameter %q", key))
+		}
+		if val == "" {
+			config[key] = params[key].Default
+		}
+	}
+	return multiErr
+}
+
+// validateParamType validates that a parameter value is parsable to its assigned type.
+func validateParamType(p Parameter, key string, value string) error {
+	switch p.Type {
+	case ParameterTypeNumber:
```

Comment:
WDYT about having different types for integers and floats?

Reply:
hmm it makes sense, a number works fine both ways, but I can see how having two types is better for the developer to add constraints on the user's configurations.
that would need a change for the connector protocol again, the number type was added recently and basically not used by anyone.. IIUC this mean we can replace it with another type without causing any backward compatibility problems. right? 
@lovromazgon WDYT?

Reply:
There are fields that can only be integers (e.g. timestamps, batch sizes, etc.), and if you by mistake enter a float, the validation will pass, even though it's not really correct.

Reply:
> this mean we can replace it with another type without causing any backward compatibility problems

Yes, as long as the "string" type stays untouched this shouldn't create any issues. Let's do it now, after the next release it won't be so easy anymore.